### DISCUSSION TITLE
Operators comprised more than one symbols

### DISCUSSION THREAD

**Discussion by nsgnkhibdk2cls0f** [orbitalquark/textadept#308] [Q&A] [answered]
On lexer file, operators is defined as a string without any delimiter between the symbols like this:

```
-- Operators.
lex:add_rule('operator', token(lexer.OPERATOR, S('~!.,:;+-*/<>=\\^|&%?()[]{}')))
``` 

Then how could we handle operators comprised more than one symbols like `>>`, `<<`, `>>>`, `<<<`, `->`, `<>`, `:=`,...?

**Comment by orbitalquark** [comment, marked as answer]
As written, lexers will match your '>>', ':=', etc. one character at a time, giving the appearance of matching distinct operators. If you want to match multiple characters at once, use ordered choice: `P('<<') + '>>' + S('+-*/'`

**Reply by nsgnkhibdk2cls0f** [reply]
Could you give me the full line as an example?

**Reply by nsgnkhibdk2cls0f** [reply]
@orbitalquark

**Reply by nsgnkhibdk2cls0f** [reply]
Yeah, I see. You have a markdown syntax error so I don't see your code formatted as code and my eyes are pretty bad. Sorry.

**Comment by paaguti** [comment]
Same principle as with multi-line comments...

On Sat, 17 Dec 2022 at 00:44, nsgnkhibdk2cls0f ***@***.***>
wrote:

> Could you give me the full line as an example?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/308#discussioncomment-4431369>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4ASQF2AIWVVEOD4I23WNT5D7ANCNFSM6AAAAAATAZOULE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Reply by nsgnkhibdk2cls0f** [reply]
Please give me an example code.
